Index: amadec/audio-dec.h
===================================================================
--- amadec/audio-dec.h	(revision 1986)
+++ amadec/audio-dec.h	(revision 1987)
@@ -197,6 +197,7 @@
     int DTSHDIEC958_PktType;
     int DTSHDPCM_SamsInFrmAtMaxSR;
     unsigned int has_video;
+	int pcm_out_count;
 };
 
 //from amcodec
Index: amadec/adec-ffmpeg-mgt.c
===================================================================
--- amadec/adec-ffmpeg-mgt.c	(revision 1986)
+++ amadec/adec-ffmpeg-mgt.c	(revision 1987)
@@ -262,10 +262,12 @@
     } else {
         adec_print("====abuf have not open!\n", val);
     }
-
-    if (am_getconfig_bool("media.arm.audio.apts_add")) {
+    if(audec->tsync_mode != TSYNC_MODE_PCRMASTER )
+    {
+    if (am_getconfig_bool("media.arm.audio.apts_add") || audec->pcm_out_count == 0) {
         offset = 0;
     }
+	}
     pts = offset;
     if (!audec->first_apts_lookup_over) {
         audec->last_valid_pts = pts;
@@ -512,6 +514,7 @@
     audec->sn_threadid = -1;
     audec->sn_getpackage_threadid = -1;
     audec->OmxFirstFrameDecoded = 0;
+	audec->pcm_out_count = 0;
     package_list_init(audec);
     while (0 != set_sysfs_int(DECODE_ERR_PATH, DECODE_NONE_ERR)) {
         adec_print("[%s %d]set codec fatal failed ! \n", __FUNCTION__, __LINE__);
@@ -1222,6 +1225,7 @@
 
                 dlen = adec_ops->decode(audec->adec_ops, outbuf, &outlen, inbuf + declen, inlen);
                 if (outlen > 0) {
+                    audec->pcm_out_count++;
                     check_audio_info_changed(audec);
                 }
                 if (outlen > AVCODEC_MAX_AUDIO_FRAME_SIZE) {
Index: amadec/adec_omx_brige.c
===================================================================
--- amadec/adec_omx_brige.c	(revision 1986)
+++ amadec/adec_omx_brige.c	(revision 1987)
@@ -189,6 +189,7 @@
             }
         }
         if (outlen > 0) {
+			audec->pcm_out_count++;
             memset(&g_AudioInfo, 0, sizeof(AudioInfo));
             g_AudioInfo.channels = (*audec->parm_omx_codec_get_Nch)(audec);
             g_AudioInfo.samplerate = (*audec->parm_omx_codec_get_FS)(audec);
